janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -367,17 +368,18 @@ String getKeyspaceName() {
 
     @VisibleForTesting
     Map<String, String> getCompressionOptions(final String name) throws BackendException {
-        KeyspaceMetadata keyspaceMetadata = this.session.getMetadata().getKeyspace(this.keyspace)
-            .orElseThrow(() -> new PermanentBackendException(String.format("Unknown keyspace '%s'", this.keyspace)));
-
-        TableMetadata tableMetadata = keyspaceMetadata.getTable(name)
-            .orElseThrow(() -> new PermanentBackendException(String.format("Unknown table '%s'", name)));
-
+        TableMetadata tableMetadata = getTableMetadata(name);
         Object compressionOptions = tableMetadata.getOptions().get(CqlIdentifier.fromCql("compression"));
-
         return (Map<String, String>) compressionOptions;
     }
 
+    @VisibleForTesting
+    Optional<String> getSpeculativeRetry(final String name) throws BackendException {
+        TableMetadata tableMetadata = getTableMetadata(name);
+        Object res = tableMetadata.getOptions().get(CqlIdentifier.fromCql("speculative_retry"));
+        return res == null ? Optional.empty() : Optional.of((String) res);
```

Comment:
nitpick:
```suggestion
        return Optional.ofNullable((String) res);
```

Reply:
Actually, it seems "99PERCENTILE" is the default value and this "res" is always non-null, so the "Optional" is probably not needed.

Reply:
Thanks Boxuan. I've updated the code.